build.gradle

```diff
@@ -44,3 +44,10 @@ test {
 checkstyle {
     toolVersion = "6.16"
 }
+
+gradle.projectsEvaluated {
+    tasks.withType(JavaCompile) {
+        // fail build on warnings, disable options complaining about Java 6 compatibility when building with JDK 7+
```

Comment:
If we are supporting JDK 6 then I'm not sure that suppressing the Java 6 compatibility warning is something that we should do. If we want to guarantee compatibility then we should set `options.bootClasspath` to point to a JDK6 `rt.jar`.

On the other hand, I realize that this means that a JDK 6 installation is required to build which might not be practical. If you think it's fine to suppress this then we should look at getting JDK 6 CI sorted.

Reply:
Fair point - I'll see if we can prioritise getting JDK 6 building on CI, or think about whether we need to support it still.